keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensorTest.java

```diff
@@ -839,9 +836,124 @@ public void youCanFixACustomValidationIssueByReplacingTheValue() {
+        DoubleTensor x = DoubleTensor.create(new double[]{1, 2, 3, 4}, new long[]{2, 2});
+        DoubleTensor summationRow = x.sum(1);
+        DoubleTensor expected = DoubleTensor.create(3, 7);
+        assertThat(summationRow, equalTo(expected));
+        assertThat(summationRow.getShape(), equalTo(expected.getShape()));
+    }
+
+    @Test
+    public void canSumOverSpecifiedDimensionOfVector() {
+        DoubleTensor x = DoubleTensor.create(1, 2, 3, 4);
+        DoubleTensor summation = x.sum(0);
+        DoubleTensor expected = DoubleTensor.scalar(10);
+        assertThat(summation, equalTo(expected));
+        assertThat(summation.getShape(), equalTo(expected.getShape()));
+    }
+
+    @Test
+    public void canDuplicateRank1() {
+        DoubleTensor x = DoubleTensor.create(1, 2);
+        assertEquals(x, x.duplicate());
+    }
+
+    @Test
+    public void canDuplicateRank0() {
+        DoubleTensor x = DoubleTensor.scalar(1.0);
+        assertEquals(x, x.duplicate());
+    }
+
+    @Test
+    public void doesDownRankOnSliceRank3To2() {
+        DoubleTensor x = DoubleTensor.create(1, 2, 3, 4, 1, 2, 3, 4).reshape(2, 2, 2);
+        TensorTestHelper.doesDownRankOnSliceRank3To2(x);
+    }
+
+    @Test
+    public void doesDownRankOnSliceRank2To1() {
+        DoubleTensor x = DoubleTensor.create(1, 2, 3, 4).reshape(2, 2);
+        TensorTestHelper.doesDownRankOnSliceRank2To1(x);
+    }
+
+    @Test
+    public void canSliceRank2() {
+        DoubleTensor x = DoubleTensor.create(new double[]{1, 2, 3, 4}, 2, 2);
+        DoubleTensor slice = x.slice(1, 0);
+        assertArrayEquals(new double[]{1, 3}, slice.asFlatDoubleArray(), 1e-10);
+    }
+
+    @Test
+    public void doesDownRankOnSliceRank1ToScalar() {
+        DoubleTensor x = DoubleTensor.create(1, 2, 3, 4).reshape(4);
+        TensorTestHelper.doesDownRankOnSliceRank1ToScalar(x);
+    }
+
+    @Test
+    public void canSliceRank1() {
+        DoubleTensor x = DoubleTensor.create(1, 2, 3, 4).reshape(4);
+        DoubleTensor slice = x.slice(0, 1);
+        assertArrayEquals(new double[]{2}, slice.asFlatDoubleArray(), 1e-10);
+    }
+
+    @Test
+    public void canConcatScalars() {
+        DoubleTensor x = DoubleTensor.scalar(2);
+        DoubleTensor y = DoubleTensor.scalar(3);
+
+        DoubleTensor concat = DoubleTensor.concat(x, y);
+        assertEquals(DoubleTensor.create(2, 3), concat);
+    }
+
+    @Test
+    public void canConcatVectors() {
+        DoubleTensor x = DoubleTensor.create(2, 3);
+        DoubleTensor y = DoubleTensor.create(4, 5);
+
+        DoubleTensor concat = DoubleTensor.concat(x, y);
+        assertEquals(DoubleTensor.create(2, 3, 4, 5), concat);
+    }
+
+    @Test
+    public void canConcatMatrices() {
+        DoubleTensor x = DoubleTensor.create(2, 3).reshape(1, 2);
+        DoubleTensor y = DoubleTensor.create(4, 5).reshape(1, 2);
+
+        DoubleTensor concat = DoubleTensor.concat(0, x, y);
```

Comment:
nit: don't need the zero here and below as isn't that the new concat default?

Reply:
0 is default in order to be nice with scalars a vectors but when dealing with matrices I wanted it to be explicit with which dimension is concatting.